src/System.Text.JsonLab/System/Text/Json/JsonReader.cs

```diff
@@ -7,6 +7,8 @@
 
 namespace System.Text.JsonLab
 {
+
+    [StructLayout(LayoutKind.Auto)]
```

Comment:
What is the benefit of adding this attribute?

Reply:
Rearranges and packs the fields in the most efficient layout; rather than the order you define them (except child structs, which are on pointer boundaries).

I'd always assumed it was the default layout as it is for classes; and you had to specify  `[StructLayout(LayoutKind.Sequential)]` if you wanted that but seem to not always be the case https://blogs.msdn.microsoft.com/seteplia/2017/09/21/managed-object-internals-part-4-fields-layout/

Not sure it would make much difference; due to the object reference (forces Auto); but was trying to find where the drop was (think its now the issue I opened)

Reply:
I haven't observed any difference in benchmarks when marking the struct layout as auto versus leaving it as default.

Reply:
Yeah, from the article it seems if there is a GC ref then it will switch to auto; and I think a Span or ArrayFormatter will count as that, so it shouldn't make any difference. Was just hoping...